demo/site/src/blocks/DamImageBlock.tsx

```diff
@@ -11,10 +11,10 @@ const DamImageBlock = withPreview(
             return <PreviewSkeleton type="media" hasContent={false} aspectRatio={aspectRatio} />;
         }
 
-        if (block.type === "pixelImage") {
-            return <PixelImageBlock data={block.props as PixelImageBlockData} aspectRatio={aspectRatio} {...imageProps} />;
+        if (block.type === "pixelImage" && "cropArea" in block.props) {
```

Comment:
This if presumably doesn't work if there's no crop area set in the PixelImageBlock, because then cropArea will be undefined: https://github.com/vivid-planet/comet/blob/main/packages/api/cms-api/src/dam/blocks/pixel-image-block-transformer.service.ts#L75

Reply:
You're right, I thought I checked, but it's indeed an optional field what about simply using `urlTemplate` which is required instead?
I think there is no need for refactoring the object's structure, if the issue can be resolved by ts :)

Reply:
`if (block.type === "pixelImage" && "urlTemplate" in block.props)`

Reply:
Sorry, reopening this conversation, but how can `urlTemplate` be undefined? It seems like it is everywhere required, the `PixelImageBlock` requires it, because it renders a `<NextImage />` where it is used as `src` prop
cf screenshot:

![image](https://github.com/user-attachments/assets/ca9be828-b5cf-4f25-b0e4-e9d082fdb17a)

If the `urlTemplate` is somehow undefined, the image will be broken anyways, right?

Reply:
PixelImageBlock has an if to verify that NextImage is only rendered if a DAM file is present: https://github.com/vivid-planet/comet/blob/main/packages/site/cms-site/src/blocks/PixelImageBlock.tsx#L18

If the DAM file is present, urlTemplate will be present as well. We could probably change the structure of the block's data to reflect this.

It should be typed as nullable here: https://github.com/vivid-planet/comet/blob/main/packages/api/cms-api/src/dam/blocks/pixel-image.block.ts#L175